modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/FormBuilder.js

```diff
@@ -51,6 +58,50 @@ export const FormBuilder = () => {
 		[initialSidebarPanels, portletNamespace]
 	);
 
+	useEffect(() => {
+		if (allowFieldSets && contentType) {
+			let globalFieldSetsPromise = [];
+
+			if (groupId) {
+				globalFieldSetsPromise = getItem(
+					`/o/data-engine/v2.0/sites/${groupId}/data-definitions/by-content-type/${contentType}`
+				);
+			}
+
+			const groupFieldSetsPromise = getItem(
+				`/o/data-engine/v2.0/data-definitions/by-content-type/${contentType}`
+			);
+
+			const fetchFieldSets = async () => {
+				try {
+					const [
+						{items: globalFieldSets = []},
+						{items: groupFieldSets = []},
+					] = await Promise.all([
+						globalFieldSetsPromise,
+						groupFieldSetsPromise,
+					]);
+
+					dispatch({
+						payload: {
+							fieldSets: [...globalFieldSets, ...groupFieldSets],
+						},
+						type: SIDEBAR.FIELDSET.UPDATE,
+					});
+				}
+				catch (error) {
+					if (process.env.NODE_ENV === 'development') {
+						console.warn(
+							`[DataEngineFormBuilder] fetchFieldSets promise rejected: ${error}`
+						);
+					}
+				}
+			};
+
+			fetchFieldSets();
+		}
+	}, [allowFieldSets, contentType, dispatch, groupId]);
+
```

Comment:
Why not move this into the FieldSet component since only it will use it, we can take advantage of the lazy call there otherwise the best place for this would be in the `useData` that entered the master recently but this blocks the rendering of the application then does not make sense.